e2e/integration_tests/graph_model_golden_tests.ts

```diff
@@ -128,23 +128,26 @@ async function expectTensorsToEqualGoldens(
   if (tensors instanceof tfc.Tensor) {
     expectTensorToEqualGolden(tensors, goldens as TensorDetail);
   } else if (Array.isArray(tensors)) {
-    const details = goldens as TensorDetail[];
     expect(Array.isArray(goldens)).toEqual(true);
+    const details = goldens as TensorDetail[];
     expect(tensors.length).toEqual(details.length);
     for (let i = 0; i < tensors.length; ++i) {
       expectTensorToEqualGolden(tensors[i], details[i]);
     }
   } else {
     const detailMap = goldens as Record<string, TensorDetail>;
+    expect(new Set(Object.keys(detailMap)))
+        .toEqual(new Set(Object.keys(tensors)));
     for (const [name, detail] of Object.entries(detailMap)) {
       expectTensorToEqualGolden(tensors[name], detail);
     }
   }
 }
 
 function isTensorDetail(x: any): x is TensorDetail {
```

Comment:
Nit: Should `x` be `unknown` instead of `any`?
```suggestion
function isTensorDetail(x: unknown): x is TensorDetail {
```

Reply:
I need to cast to something to access `x.shape` and others, or it would throw an error. Any is simpler here.